performance/plugins/extrinsicSwap.ts

```diff
@@ -29,65 +37,63 @@ export class ExtrinsicSwap extends performanceTestItem {
       return values.map((val) => val.toNumber());
     });
     //create the pool
+    //
     await this.createPoolIfMissing(
       new BN(this.tokens[0]),
       new BN(this.tokens[1]),
-      nodes
+      testParams.nodes
     );
-    await this.mintTokensToUsers(numberOfThreads, nodes);
+    await this.mintTokensToUsers(testParams.threads, testParams.nodes);
     await this.mintERC20TokensToUsers(this.tokens, this.mgaNodeandUsers);
     console.info(`Setup Done!`);
     tokens = this.tokens;
     return true;
   }
   async act(testParams: TestParams): Promise<boolean> {
-    let keepCalling = true;
-    setTimeout(function () {
-      keepCalling = false;
-      //duration in minutes, transform to millisecs.
-    }, testParams.duration * 60 * 1000);
-    while (keepCalling) {
-      const preSetupThreads = await preGenerateTransactions(
-        testParams,
-        this.mgaNodeandUsers,
-        createAndSignSwaps,
-        { testParams }
-      );
-      console.info(`running Txs..`);
-      await runTransactions(testParams, preSetupThreads);
-    }
-    console.info(`Done running Txs!`);
+    await super.act(testParams);
+    const generator = (
+      sdk: Mangata,
+      users: { nonce: BN; keyPair: KeyringPair }[],
+      thread: number,
+      offset: BN
+    ) => {
+      return createAndSignSwaps(this.isBuy, sdk, users, thread, offset);
+    };
+    await runTransactions(this.mgaNodeandUsers, testParams, generator);
```

Comment:
each perofrmance test item needs just to provide generator functor, single call to generator returns single signed tx read y to be sent